packages/ccip-svm/src/__tests__/evmExtraArgsV2.test.ts

```diff
@@ -0,0 +1,179 @@
+
+            // Should not show any warning for recommended value
+            expect(consoleSpy).not.toHaveBeenCalled()
+
+            const expected = "0x181dcf1000000000000000000000000000000000000000000000000000000000000186a00000000000000000000000000000000000000000000000000000000000000001";
+
+            expect(encoded).toBeInstanceOf(Uint8Array)
+            expect(encoded.length).toBe(68)
+            expect(encoded).toEqual(new Uint8Array(Buffer.from(expected.slice(2), 'hex')))
+        })
+
+        it('should encode EVM extra args with allowOutOfOrderExecution=false and show warning', () => {
+            const original = {
+                gasLimit: 100000n,
+                allowOutOfOrderExecution: false
+            }
+
+            const encoded = encodeEVMExtraArgsV2(original)
+
+            // Should show warning when user overrides to false
+            expect(consoleSpy).toHaveBeenCalledWith(
+                `Warning: Setting allowOutOfOrderExecution to false is not recommended for EVM destinations.`
+            )
+
+            const expected = "0x181dcf1000000000000000000000000000000000000000000000000000000000000186a00000000000000000000000000000000000000000000000000000000000000000";
+
+            expect(encoded).toBeInstanceOf(Uint8Array)
+            expect(encoded.length).toBe(68)
+            expect(encoded).toEqual(new Uint8Array(Buffer.from(expected.slice(2), 'hex')))
+        })
+
+        it('should handle token-only transfers (gasLimit = 0)', () => {
+            const original = {
+                gasLimit: 0n,
+                allowOutOfOrderExecution: true
+            }
+
+            const encoded = encodeEVMExtraArgsV2(original)
+
+            const expected = "0x181dcf1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001";
+
+            expect(consoleSpy).not.toHaveBeenCalled()
+            expect(encoded).toBeInstanceOf(Uint8Array)
+            expect(encoded.length).toBe(68)
+            expect(encoded).toEqual(new Uint8Array(Buffer.from(expected.slice(2), 'hex')))
+        })
+
+        it('should include correct tag at the beginning', () => {
+            const encoded = encodeEVMExtraArgsV2({ gasLimit: 0n })
+
+            // Extract first 4 bytes and convert to uint32 big-endian
+            const tag = new DataView(encoded.buffer, encoded.byteOffset, 4).getUint32(0, false)
+            expect(tag).toBe(EVM_EXTRA_ARGS_V2_TAG)
+            expect(tag).toBe(0x181dcf10)
+        })
+    })
+
+    describe('decodeEVMExtraArgsV2', () => {
+        it('should decode EVM extra args correctly with default values', () => {
+            const original = {
+                gasLimit: 200000n,
+                allowOutOfOrderExecution: true
+            }
+
+            const encoded = encodeEVMExtraArgsV2(original)
+            const decoded = decodeEVMExtraArgsV2(encoded)
+
+            expect(decoded.gasLimit).toBe(original.gasLimit)
```

Comment:
does toBe also assert type? or would we need to explicitly assert gasLimit is BigInt?